labs/opltestiso/ee/main.c

```diff
@@ -0,0 +1,297 @@
+    sceCdSearchFile(&fp, filename);
+
+    clk_start = clock();
+    sceCdStStart(fp.lsn, &mode);
+    while (size_left) {
+        int rv = sceCdStRead(sectors, eebuffer, STMBLK, &error);
+        if (rv != sectors) {
+            PRINTF("\t\t- sceCdStRead = %d error = %d\n", rv, error);
+            //    break;
+        }
+        if (error != SCECdErNO) {
+            PRINTF("\t\t- ERROR %d\n", error);
+            break;
+        }
+        size_left -= rv * 2048;
+    }
+    sceCdStStop();
+    clk_end = clock();
+
+    print_speed(clk_start, clk_end, total_size - size_left, block_size);
+
+    free(eebuffer);
+    SifFreeIopHeap(iopbuffer);
+}
+
+//--------------------------------------------------------------
+void test_read_stream(const char *filename)
+{
+    unsigned int block_size;
+
+    PRINTF("\t\tStart reading file %s streaming mode:\n", filename);
+    for (block_size = STREAM_BLOCK_SIZE_MIN; block_size <= STREAM_BLOCK_SIZE_MAX; block_size *= 2)
+        test_read_stream_1(filename, block_size, FILE_SIZE);
+}
+
+//--------------------------------------------------------------
+void test_free_iop_ram()
+{
+    s32 freeram = 0;
+    s32 usedram = 0;
+
+    // Check free IOP RAM
+    if (SifExecModuleBuffer(freeram_irx, size_freeram_irx, 0, NULL, NULL) < 0)
+        PRINTF("\t\tcould not load %s\n", "freeram.irx");
+
+    smem_read((void *)(1 * 1024 * 1024), &freeram, 4);
+    usedram = (2 * 1024 * 1024) - freeram;
+    PRINTF("\t\tFree IOP RAM: %4dKiB (%7d bytes)\n", freeram / 1024, freeram);
+    PRINTF("\t\tUsed IOP RAM: %4dKiB (%7d bytes)\n", usedram / 1024, usedram);
+}
+
+//--------------------------------------------------------------
+void print_header()
+{
+    PRINTF("\n\n\n");
+    PRINTF("\t\tOPL tester v0.5\n");
+}
+
+//--------------------------------------------------------------
+#define NDW -20
+#define HDW -9
+void print_iop_modules()
+{
+    smod_mod_info_t info;
+    smod_mod_info_t *curr = NULL;
+    char sName[21];
+    int iPage = 1;
+    int iModsPerPage = 0;
+    int rv = -1;
+    int i;
+    u32 txtsz_total = 0;
+    u32 dtasz_total = 0;
+    u32 bsssz_total = 0;
+
+    while (rv != 0) {
+        scr_clear();
+        print_header();
+        PRINTF("\t\tLoaded Modules page %d:\n", iPage);
+        PRINTF("\t\t%*s   %*s %*s %*s %*s\n", NDW, "name", HDW, "txtst", HDW, "txtsz", HDW, "dtasz", HDW, "bsssz");
+        PRINTF("\t\t-----------------------------------------------------------\n");
+        while ((rv = smod_get_next_mod(curr, &info)) != 0) {
+            smem_read(info.name, sName, 20);
+            sName[20] = 0;
+            txtsz_total += info.text_size;
+            dtasz_total += info.data_size;
+            bsssz_total += info.bss_size;
+
+            PRINTF("\t\t%*s | 0x%5x | 0x%5x | 0x%5x | 0x%5x\n", NDW, sName, info.text_start, info.text_size, info.data_size, info.bss_size);
+            curr = &info;
+
+            iModsPerPage++;
+            if ((iModsPerPage >= 16) && (rv != 0)) {
```

Comment:
The `rv != 0` check is already handled by the while condition on line 193

Reply:
Yes, I saw in codacy also but didn't get around to fixing it.
Force pushed the fix just now.